hazelcast/config.py

```diff
@@ -1369,19 +1369,15 @@ def use_public_ip(self, value):
             raise TypeError("use_public_ip must be a boolean")
 
     @property
-    def compact_serializers(self) -> typing.Dict[typing.Type, CompactSerializer]:
+    def compact_serializers(self) -> typing.List[CompactSerializer]:
         return self._compact_serializers
 
     @compact_serializers.setter
-    def compact_serializers(self, value: typing.Dict[typing.Type, CompactSerializer]) -> None:
-        if isinstance(value, dict):
-            for clazz, serializer in value.items():
-                if not isinstance(clazz, type):
-                    raise TypeError("Keys of compact_serializers must be classes")
```

Comment:
We are losing the validation of classes here. Can we somehow check that getClass method exists in the serializer and it returns a class and nothing else. Does instanceof check takes care of the existence of methods ? I am sure it does not  handle the validation of the return values of getClass and getTypeName :). 

I know the `get_type_name` validation was nonexistent before as well but I think we can add it. In Node.js, because of fail fast goal of ours, I am calling the methods and checking the return value's type. See https://github.com/hazelcast/hazelcast-nodejs-client/blob/801696baf6492475ad1ddf664356a9707a0bf092/src/config/ConfigBuilder.ts#L512 and https://github.com/hazelcast/hazelcast-nodejs-client/blob/801696baf6492475ad1ddf664356a9707a0bf092/src/config/ConfigBuilder.ts#L517

Reply:
Since the user has to derive their class from `CompactSerializer)`, the class will have its `get_class` method.

Reply:
What about return value validation?

Reply:
That's something to be caught by the type checker.

Reply:
However not everyone uses a type checker in python. 

Think of Node.js client. For non-ts users we do a runtime check even we don't have to. Isn't this what we want to do, failing fast? 

In java, we the compilation step validates the type. 

If we have the goal of failing fast given a  wrong configuration, we should do validate here.

Reply:
We will discuss later how much to validate. Skipping now.